apps/api/src/app/events/services/calculate-delay/delay.service.ts

```diff
@@ -0,0 +1,65 @@
+import { DigestUnitEnum, DelayTypeEnum } from '@novu/shared';
+
+import { Logger } from '@nestjs/common';
+import { ApiException } from '../../../shared/exceptions/api.exception';
+import { differenceInMilliseconds } from 'date-fns';
+import { NotificationStepEntity } from '@novu/dal';
+
+export class DelayService {
+  calculateDelay(step: NotificationStepEntity, payload: any, overrides: any): number {
```

Comment:
What do you think about wrapping the input here in an object?
In addition, get rid of the 'any' types.